cql3/expr/expression.cc

```diff
@@ -2300,5 +2300,28 @@ type_of(const expression& e) {
     }, e);
 }
 
+bool is_single_column_restriction(const expression& e) {
+    if (find_in_expression<unresolved_identifier>(e, [](const auto&) {return true;})) {
+        on_internal_error(expr_logger, format("is_single_column_restriction - expression is not prepared: {}", e));
```

Comment:
I hope it dies soon, since we ought not to do such checks in runtime.

Reply:
I was worried about the case where someone unknowingly uses this function on an unprepared expression.
Ignoring `unresolved_identifiers` will cause the function to give the wrong answers silently which could be hard to debug.
This could be a bit slower, but I think this function is only used during query analysis, so it shouldn't matter much,